From f56e59712d8708c74a4cf66d24b5703fc16b22ae Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 1 Sep 2022 11:31:48 -0700
Subject: [PATCH] configure: Check for request_init API when probing for
 libwrap

checking for main() is not right check, since this function is not part
of libwrap but the app. Newer autocof and toolchain may fail

Upstream-Status: Submitted [https://github.com/cminyard/gensio/pull/48]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 90cfcb6b..1c06d918 100644
--- a/configure.ac
+++ b/configure.ac
@@ -109,7 +109,7 @@ AC_ARG_WITH(tcp-wrappers,
 if test "$tcp_wrappers" != "no"
 then
   AC_CHECK_HEADERS([tcpd.h])
-  AC_CHECK_LIB(wrap,main)
+  AC_CHECK_LIB(wrap,request_init)
 fi
 
 AC_ARG_ENABLE([doc],
-- 
2.37.3

